### ISSUE TITLE
Importing from module with same name as tool module causes ambiguity

### ISSUE THREAD

**Issue by clubby789** [rust-lang/rust#111135]
I tried this code:

```rust
mod rustfmt { pub fn skip() {} }

use rustfmt::skip;
fn main() {
  skip()
}
```

I expected to see this happen: No error, as the `rustfmt` tool module cannot be imported from, so the path should resolve to the module.

Instead, this happened: 
```
error[E0659]: `rustfmt` is ambiguous
 --> poc.rs:3:5
  |
3 | use rustfmt::skip;
  |     ^^^^^^^ ambiguous name
  |
  = note: ambiguous because of multiple potential import sources
  = note: `rustfmt` could refer to a tool module
note: `rustfmt` could also refer to the module defined here
 --> poc.rs:1:1
  |
1 | mod rustfmt { pub fn skip() {} }
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: use `crate::rustfmt` to refer to this module unambiguously
```

This would make adding new tool namespaces (such as the proposed [RFC 3368](https://github.com/rust-lang/rfcs/pull/3368) a breaking change.

### Meta

Present on stable, nightly and master.

@rustbot label +A-resolve

**Comment by petrochenkov** [comment]
This is an instance of https://github.com/rust-lang/rust/issues/56414.

**Comment by clubby789** [comment]
Should this be closed then?

**Comment by petrochenkov** [comment]
As you want, in any case this can be used as one more motivational case for removing the check from #56414.
